features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/RumSessionEndedMetricTest.kt

```diff
@@ -0,0 +1,409 @@
+import com.datadog.tools.unit.extensions.TestConfigurationExtension
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.json.JSONObject
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+import java.util.concurrent.TimeUnit
+import kotlin.random.Random
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+    ExtendWith(TestConfigurationExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+class RumSessionEndedMetricTest {
+
+    @StringForgery
+    private lateinit var fakeSessionId: String
+
+    @StringForgery
+    private lateinit var fakePrecondition: String
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    @Test
+    fun `M metric type W creating metric`() {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakePrecondition)
+
+        // When
+        val attributes = sessionEndedMetric.toMetricAttributes()
+
+        // Then
+        Assertions.assertEquals(attributes[SessionEndedMetric.METRIC_TYPE_KEY], SessionEndedMetric.METRIC_TYPE_VALUE)
+    }
+
+    @Test
+    fun `M process type W creating metric`() {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakePrecondition)
+
+        // When
+        val attributes = sessionEndedMetric.toMetricAttributes()
+
+        // Then
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val processType = rse[SessionEndedMetric.PROCESS_TYPE_KEY] as String
+        Assertions.assertEquals(SessionEndedMetric.PROCESS_TYPE_VALUE, processType)
+    }
+
+    @Test
+    fun `M precondition W creating metric`() {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakePrecondition)
+
+        // When
+        val attributes = sessionEndedMetric.toMetricAttributes()
+
+        // Then
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val precondition = rse[SessionEndedMetric.PRECONDITION_KEY] as String
+        Assertions.assertEquals(fakePrecondition, precondition)
+    }
+
+    @Test
+    fun `M init attributes W creating empty metric`() {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakePrecondition)
+
+        // When
+        val attributes = sessionEndedMetric.toMetricAttributes()
+
+        // Then
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+
+        // Assertions.assertTrue() // todo with duration
```

Comment:
Don't leave dangling `TODO` - it is either should be addressed to should be backed by the JIRA ticket reference. Here I think it should be handled as a part of this PR.

Reply:
it's a forgetten todo, now fixed now